spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/pubsublite/GcpPubSubLiteAutoConfiguration.java

```diff
@@ -0,0 +1,93 @@
+
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.cloud.pubsublite.CloudZone;
+import com.google.cloud.pubsublite.cloudpubsub.FlowControlSettings;
+import com.google.cloud.spring.autoconfigure.core.GcpContextAutoConfiguration;
+import com.google.cloud.spring.autoconfigure.pubsublite.GcpPubSubLiteProperties.FlowControl;
+import com.google.cloud.spring.core.DefaultCredentialsProvider;
+import com.google.cloud.spring.core.GcpProjectIdProvider;
+import com.google.cloud.spring.pubsub.support.PublisherFactory;
+import com.google.cloud.spring.pubsub.support.StreamingSubscriberFactory;
+import com.google.cloud.spring.pubsublite.support.DefaultPublisherFactory;
+import com.google.cloud.spring.pubsublite.support.DefaultSubscriberFactory;
+import com.google.cloud.spring.pubsublite.support.PublisherFactorySettings;
+import com.google.cloud.spring.pubsublite.support.SubscriberFactorySettings;
+
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration(proxyBeanMethods = false)
+@AutoConfigureAfter(GcpContextAutoConfiguration.class)
+@ConditionalOnProperty(value = "spring.cloud.gcp.pubsublite.enabled", matchIfMissing = true)
+@ConditionalOnClass(PublisherFactorySettings.class)
+@EnableConfigurationProperties(GcpPubSubLiteProperties.class)
+public class GcpPubSubLiteAutoConfiguration {
+	private final GcpPubSubLiteProperties properties;
+
+	private final GcpProjectIdProvider finalProjectIdProvider;
+
+	private final CredentialsProvider finalCredentialsProvider;
+
+	public GcpPubSubLiteAutoConfiguration(GcpPubSubLiteProperties properties,
+			GcpProjectIdProvider gcpProjectIdProvider,
+			CredentialsProvider credentialsProvider) throws IOException {
+		this.properties = properties;
+		this.finalProjectIdProvider = (properties.getProjectId() != null)
+				? properties::getProjectId
+				: gcpProjectIdProvider;
+		this.finalCredentialsProvider = properties.getCredentials().hasKey()
+				? new DefaultCredentialsProvider(properties)
+				: credentialsProvider;
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public PublisherFactory defaultPublisherFactory() {
+		PublisherFactorySettings settings = PublisherFactorySettings.newBuilder()
+				.setLocation(CloudZone.parse(properties.getLocation()))
+				.setProjectIdProvider(finalProjectIdProvider)
+				.setCredentialsProvider(finalCredentialsProvider)
+				.build();
+		return new DefaultPublisherFactory(settings);
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public StreamingSubscriberFactory defaultSubscriberFactory() {
```

Comment:
What about `PubSubSubscriberOperations`? Can that be supported with Pub/Sub Lite?

Reply:
There is no synchronous pull for Pub/Sub Lite, only the streaming pull model is supported. This is the one method that can actually be supported: I've split out this operation to a sub-interface, PubSubStreamingSubscriberOperations, but I believe this should not be a breaking change as written.